src/init.cpp

```diff
@@ -223,9 +223,6 @@ void Shutdown()
         if (pcoinsTip != NULL) {
             FlushStateToDisk();
         }
-        // Flush the wallet witness cache to disk (no longer done by FlushStateToDisk())
-        GetMainSignals().SetBestChain(chainActive.GetLocator());
```

Comment:
Unfortunately, it's not safe to flush the witness data on clean shutdown; removing this restores the code to what it was for years until the recent PR #4627. The effect of removing this is that the last 10 minutes of blocks need to be rescanned even after a clean shutdown. This takes only a few seconds.

The reason this isn't safe is that there is no guarantee that the locator used here (part of the flush data), which is based on the latest block, is consistent with the witness data. The heights of the witness data and the locator must be the same. The fix this PR makes to wallet thread (`ChainTipAdded()`) now guarantees that, and that's really the only place it can be guaranteed.